addons/source-python/docs/source-python/source/developing/module_tutorials/menus.rst

```diff
@@ -0,0 +1,116 @@
+Menus
+======
+
+This page contains tutorials about the :mod:`menus.radio` package.
+
+
+Creating PagedMenu
+--------------------------
+
+This is a simple example to create paged menu
+
+.. code-block:: python
+
+    from commands.say import SayCommand
+
+    from menus import PagedMenu
+    from menus import PagedOption
+
+    from players.entity import Player
+
+
+    # Register menu command
+    @SayCommand(['menu', '/menu', '!menu'])
+    def say_command(command, index, teamonly):
+        # Send the menu
+        build_paged_menu(index)
+        return False
+
+    def build_paged_menu(index):
+        # Create the Paged Menu and set title to My Menu
+        menu = PagedMenu(title='My Menu')
+        # Add options from 1 to 20
+        for i in range(1, 20):
+            menu.append(PagedOption(f'{i}', i))
+        menu.select_callback=my_select_callback
+        menu.send(index)
+
+    def my_select_callback(menu, index, option):
+        # Create player object from index
+        player = Player(index)
+        # Print the player name who have selected a menu option
+        print(f'{player.name} have selected a menu option!')
+
+
+Creating SimpleMenu
+--------------------------
+
+This is a simple example to create simple menu
+
+.. code-block:: python
+
+    from commands.say import SayCommand
+
+    from menus import SimpleMenu
+    from menus import SimpleOption
+    from menus import Text
+
+    from messages import SayText2
+
+    from players.entity import Player
+
+    # Register menu command
+    @SayCommand(['menus', '/menus', '!menus'])
+    def say_menus_command(command, index, teamonly):
+        # Send the menu
+        build_simple_menu(index)
+        return False
+
+    def build_simple_menu(index):
+        # Create the SimpleMenu
+        menu = SimpleMenu
+        # Add in menu text
+        menu.append(Text('Do you accept the rules?'))
+        menu.append(Text(' '))
+        # Add in menu options
+        menu.append(SimpleOption(1, 'Yes', 'yes'))
+        menu.append(SimpleOption(2, 'No', 'no'))
+        menu.select_callback=my_menu_select_callback
+        menu.send(index)
+
+    def my_menu_select_callback(menu, index, option):
+        choice = option.value
+        # Create player object from index
+        player = Player(index)
+        # Player did select yes option
+        if choice == 'yes':
+            SayText2('Thank you for accepting the rules!').send(index)
+        # Player selected no option
+        else:
+            player.kick('You have to accept the rules!')
```

Comment:
This should rather be:

```python
    def my_menu_select_callback(menu, index, option):
        if option.value == 'yes':
            SayText2('Thank you for accepting the rules!').send(index)
        # Player selected no option
        else:
            Player(index).kick('You have to accept the rules!')
```

As there is no reason to locally assign values to your frame unless you use them multiple times. Moreover, there is no reason to instantiate a `Player` unless your condition is not met so that is not something you should do otherwise.

Reply:
I have now updated the wiki tutorial based of your comments.
Should in wiki menu tutorial include using WeaponClassIter for making buymenu?
As example looping the weapons showing for their name and their price and it would check player can afford to buy it